### PR TITLE
tdvp: allow changing solver without reconstructing

### PR DESCRIPTION
**Pull request by PhilipVinc**
Useful to change the dt in the midst of the integration

### CODE REVIEW

File: netket/experimental/driver/tdvp.py [hunk 1]
```diff
@@ -173,6 +166,41 @@ def generator(self) -> Callable:
         """
         return self._generator
 
+    @property
+    def error_norm(self) -> Callable:
+        """
+        Returns the Callable function computing the error of the norm used for adaptive
+        timestepping by the integrator.
+        """
+        if self._error_norm == "euclidean_norm":
```
Comment (femtobit) [inline netket/experimental/driver/tdvp.py:175]: Is it necessary to run the full logic here every time `error_norm` is called? It doesn't depend on anything that changes dynamically, if I'm not mistaken.

Comment (PhilipVinc) [reply netket/experimental/driver/tdvp.py:175]: We only call it once when constructing the integrator, no?

Comment (PhilipVinc) [reply netket/experimental/driver/tdvp.py:175]: otherwise yeah, i think you're right we're better off put this logic in the setter

Comment (PhilipVinc) [reply netket/experimental/driver/tdvp.py]: done

**Review by femtobit — COMMENTED** [review]
Looks good to me, assuming the tests pass. Maybe the test can be slightly extended:

File: test/dynamics/test_driver.py [hunk 2]
```diff
@@ -240,3 +240,22 @@ def test_run_twice():
...
+    driver.run(0.03)
+    np.testing.assert_allclose(driver.t, 0.03)
+
+    integrator = nkx.dynamics.Euler(dt=0.05)
+    driver.integrator = integrator
+    np.testing.assert_allclose(driver.t, 0.03)
+    np.testing.assert_allclose(driver.dt, 0.05)
+
+    driver.run(0.1)
+    np.testing.assert_allclose(driver.t, 0.13)
```
Comment (femtobit) [inline test/dynamics/test_driver.py:261]: I think it would be good to add a few lines testing a change of the error norm too.

**Review by femtobit — APPROVED** [review]

**Pull request merged**